### DISCUSSION TITLE
zooming windows

### DISCUSSION THREAD

**Discussion by tve** [hyprwm/Hyprland#3467] [Q&A]
I'd like to make a layout where non-focus windows are zoomed 0.5x, i.e. they're smaller than normal. Something similar to qtile's zoomy layout (https://qtile-dmp.readthedocs.io/en/latest/manual/ref/layouts.html#zoomy) except that I'd actually like to shrink the content, not just resize the windows. Is this possible in hyprland? If not, is this actually doable technically?

**Comment by justyn** [comment]
I was also thinking of this, it would be really useful on a small screen.

I'm not sure though whether it is possible without changes to Hyprland, because I don't think there is a way to set per-window scaling?

Commit b05ff89c76b2c9454c76daf2120f4ccb076a018d added a cursor zoom, but I think this is just drawing a box over the existing content (see OpenGL.cpp changes).

**Comment by AF0BL** [comment]
I came here from #3861 , I am happy with my scaling overall and don't want to change it globally, but I have a couple applications that have tiny UI's that I want to be bigger. I would prefer per-window scaling, per-workspace scaling would also solve my issue.

**Comment by charnam** [comment]
I would personally like to see if it is possible to add a windowrule to change the content scaling of windows. If not, some sort of "pseudo-content-scaling" where an application window's internal perceived width is double, yet visually rendered smaller on-screen. It could be used via "windowscale 0.5" or "renderwidth 1000" or something similar.

This would be extremely useful for the master layout, where a focused window on the left is visually scaled at 1x, while other windows on the right could be scaled at 0.5x in order to reduce the chances of the dreaded "tiling window manager has broken window content due to autotiling" issue, which I seem to encounter frequently while attempting to use `master`.

A perceived width increase with a maintained visual downscale window rule, (for example, used with non-focused or non-master windows) would be absolutely amazing. Though initially it may seem that this would hinder performance due to rendering large windows and downscaling them, this could simply be stated as a warning to those who use the rule. I've been considering this since I first used Hyprland and am surprised it has not been considered as a feature yet.

If I am dumb, or not realizing that there's already a way to do this, please let me know.

Edit: Wayland per-output scaling could also be used to achieve this, but personally I feel that if that's going to be implemented as a solution here, then there should at least be a fallback in place to instead render initially and downscale, if desired. Some applications do not handle per-output scaling properly, and I'd like if we could see consistent behavior. Though, do what you wish, of course.
